pkg/docker/config/config.go

```diff
@@ -215,22 +216,46 @@ func getAuthFilePaths(sys *types.SystemContext, homeDir string) []authPath {
 // helpers with falling back to using either auth.json
 // file or .docker/config.json, including support for OAuth2 and IdentityToken.
 // If an entry is not found, an empty struct is returned.
+//
+// Deprecated: GetCredentialsForRef should be used in favor of this API
+// because it supports the registry path lookup feature.
 func GetCredentials(sys *types.SystemContext, registry string) (types.DockerAuthConfig, error) {
-	return getCredentialsWithHomeDir(sys, registry, homedir.Get())
+	return getCredentialsWithHomeDir(sys, nil, registry, homedir.Get())
 }
 
-// getCredentialsWithHomeDir is an internal implementation detail of GetCredentials,
-// it exists only to allow testing it with an artificial home directory.
-func getCredentialsWithHomeDir(sys *types.SystemContext, registry, homeDir string) (types.DockerAuthConfig, error) {
+// GetCredentialsForRef returns the registry credentials stored in the
+// registry-specific credential helpers or in the default global credentials
+// helpers with falling back to using either auth.json
+// file or .docker/config.json, including support for OAuth2 and IdentityToken.
+// Compared to GetCredentials, this function accepts a full ref, which will
+// be used for supporting multiple paths when looking up the credentials
```

Comment:
A little more polishing: don’t talk about “registry credentials”, the “compared to GetCredentials” is not something readers care about.

Note to self: propose an alternative.

Reply:
My mistake, “registry credentials” is actually correct.

----

How about much simpler
```go
// GetCredentialsForRef returns the registry credentials necessary for
// accessing ref on the registry ref points to,
// appropriate for sys and the users’ configuration.
// If an entry is not found, an empty struct is returned.
```
? Or maybe expand this to “configuration (based on configuration in user’ home directory, global configuration files and elsewhere)” but that, vague enough not to commit to specifics, feels vague enough to not be useful to a caller.

----

Alternatively, we don’t need spend time in _this_ PR polishing the pre-existing documentation from `GetCredentials`:
```go
// GetCredentialsForRef returns the registry credentials necessary for
// accessing ref on the registry ref points to, stored in the
// registry-specific credential helpers or in the default global credentials
// helpers with falling back to using either auth.json
// file or .docker/config.json, including support for OAuth2 and IdentityToken.
// If an entry is not found, an empty struct is returned.
```

Reply:
Thank you for the suggestion, I decided to use your simple version (the first one).